pandas/core/dtypes/dtypes.py

```diff
@@ -1,5 +1,7 @@
 """ define extension dtypes """
+import builtins
```

Comment:
So I found some more information on this error:

https://github.com/python/mypy/issues/1775

The suggested approach there is to alias the type rather than import builtins.

However, this comment by Guido might also be of importance:

https://github.com/python/mypy/issues/1775#issuecomment-311120081

His example cites a method where a subsequent annotation mangles a method with the name of `bytes` with the builtin. I imagine this is the same with the static variable (if you could confirm would be great) so we would have to be pretty careful when typing this module with `str` and `type` to avoid nuanced errors

Reply:
I can confirm that within the body of the classes, the class variables `str` and `type` are clobbering the builtins. So, e.g, in the class itself (but not in its methods, which each have their own scope) `isintance("asdf", str)` will throw a `TypeError` because the second argument isn't a type or tuple of types. Likewise, `isinstance(int, type)` in that location will throw the same error.

Within the methods, things work normally, see e.g., https://github.com/pandas-dev/pandas/blob/d8af65086c085bb7d22d920fe52fc3d714fbb571/pandas/core/dtypes/dtypes.py#L541